sdk/src/transaction_context.rs

```diff
@@ -682,19 +810,101 @@ impl<'a> BorrowedAccount<'a> {
             )
             .unwrap_or_default()
     }
+
+    /// Returns true if the owner of this account is the current `InstructionContext`s last program (instruction wide)
+    pub fn is_owned_by_current_program(&self) -> bool {
+        self.instruction_context
+            .get_last_program_key(self.transaction_context)
+            .map(|key| key == self.get_owner())
+            .unwrap_or_default()
+    }
+
+    /// Returns an error if `get_data_mut`, `set_data` or `set_data_length` would fail
+    pub fn is_data_mutable(&self, new_length: usize) -> Result<(), InstructionError> {
+        if !self
+            .transaction_context
+            .enable_early_verification_of_account_modifications
+        {
+            return Ok(());
+        }
+        let not_owner = !self.is_owned_by_current_program();
+        if new_length != self.get_data().len() {
+            // Accounts data meter
+            let total_resize_delta = self
+                .transaction_context
+                .total_resize_delta
+                .borrow()
+                .saturating_add((new_length as i64).saturating_sub(self.get_data().len() as i64));
+            if total_resize_delta > self.transaction_context.total_resize_limit as i64 {
```

Comment:
This check should be behind the `cap_accounts_data_len` feature

Reply:
@brooksprumo ATM, if this feature were to be enabled, the account data allocation check inside the transaction processing would be gone. See the second last commit "Relaxes AccountDataMeter related test cases". So, I think there should be some replacement in another PR, and the feature gate issue needs to keep track of it.